src/compiler/checker.ts

```diff
@@ -6699,7 +6699,14 @@ namespace ts {
                 let hasThisParameter: boolean;
                 const iife = getImmediatelyInvokedFunctionExpression(declaration);
                 const isJSConstructSignature = isJSDocConstructSignature(declaration);
-                const isUntypedSignatureInJSFile = !iife && !isJSConstructSignature && isInJavaScriptFile(declaration) && !hasJSDocParameterTags(declaration);
+                const isUntypedSignatureInJSFile = !iife &&
+                    !isJSConstructSignature &&

```

Comment:
`!isJSConstructSignature` is redundant if we test `declaration.kind !== SyntaxKind.FunctionType` anyway.

Reply:
Nice catch. Removed.